### ISSUE TITLE
gici_main: run GIC-“segment fault”

### ISSUE THREAD

**Issue by layfolk1999** [chichengcn/gici-open#6]
[ 96%] Building CXX object CMakeFiles/gici.dir/src/fusion/multisensor_estimating.cpp.o
[ 96%] Building CXX object CMakeFiles/gici.dir/src/fusion/rtk_imu_camera_rrr_estimator.cpp.o
[ 97%] Building CXX object CMakeFiles/gici.dir/src/fusion/rtk_imu_tc_estimator.cpp.o
[ 97%] Building CXX object CMakeFiles/gici.dir/src/fusion/spp_imu_camera_rrr_estimator.cpp.o
[ 98%] Building CXX object CMakeFiles/gici.dir/src/fusion/spp_imu_tc_estimator.cpp.o
[ 99%] Linking CXX shared library libgici.so
[ 99%] Built target gici
Scanning dependencies of target gici_main
[ 99%] Building CXX object CMakeFiles/gici_main.dir/src/gici_main.cpp.o
[100%] Linking CXX executable gici_main
/usr/bin/ld: warning: libglog.so.0, needed by libgici.so, may conflict with libglog.so.1
[100%] Built target gici_main
liyi@liyi:~/code/gici-open-master/build$ ./gici_main ../option/pseudo_real_time_estimation_RTK.yaml 
段错误

<img width="956" alt="微信截图_20230711202938" src="https://github.com/chichengcn/gici-open/assets/139234731/70b08f15-3930-424f-a0ae-cbddb0689f1f">

Could you tell me how to solve this problem? Thanks！

**Comment by chichengcn** [comment]
It is so weird. Commonly, GICI would not terminate with segment fault without any error message presented. 

It looks like an initialization-stage error,  meaning the code did not run to line 80 in gici_main.cpp. You can add some breakpoints in debug mode or just add some std::cerr or std::cout at the lines before line 80 in gici_main.cpp. And then tell me which line causes the segment fault.

**Comment by JINGERGER** [comment]
I got the same issue yesterday. You have to make sure that your system only have one libglog.so which used by your project.I deleted other old libglogs and add some softlinks pointing a same libglog which links to the project.And it works.

**Comment by layfolk1999** [comment]
Hello, @JINGERGERJ. About libglog, there are 3 files in "usr/local/lib", such as libglog.so, libglog.so.0.6.0 and libglog.so.1, the linking order is "libglog.so->libglog.so.1->libglog.so.0.6.0". I removed libglog.so.1 by 'rm' command and linked libglog.so to libglog.so.0.6.0, still there is warning "warning: libglog.so.0, needed by libgici.so, may conflict with libglog.so.1". I entered the command "locate I entered the command "locate libglog | grep usr" and found that libglog.so.1 still exists in the directory "usr/local/lib". 

<img width="1279" alt="微信截图_20230712114043" src="https://github.com/chichengcn/gici-open/assets/139234731/740047ce-3869-4673-b2c1-e4f3c75bdd56">

Have you encountered this problem?

**Comment by zzzzyp-sgg** [comment]
> Hello, @JINGERGERJ. About libglog, there are 3 files in "usr/local/lib", such as libglog.so, libglog.so.0.6.0 and libglog.so.1, the linking order is "libglog.so->libglog.so.1->libglog.so.0.6.0". I removed libglog.so.1 by 'rm' command and linked libglog.so to libglog.so.0.6.0, still there is warning "warning: libglog.so.0, needed by libgici.so, may conflict with libglog.so.1". I entered the command "locate I entered the command "locate libglog | grep usr" and found that libglog.so.1 still exists in the directory "usr/local/lib".
> 
> <img alt="微信截图_20230712114043" width="1279" src="https://user-images.githubusercontent.com/139234731/252851141-740047ce-3869-4673-b2c1-e4f3c75bdd56.png">
> Have you encountered this problem?

Actullay I encountered a similar problem when I try to run the example data, and I shared my solution in my blog:[glog problem](https://blog.csdn.net/weixin_45432823/article/details/131542545)
Please check whether you have ever installed the "glog" library using the "apt-get install" command. And the easiest method is to delete all "glog" and "gflags" and re-install them. But you have to watch out for other libraries linking to them.
Hope this helps.

**Comment by JINGERGER** [comment]
Try to remove the one in /usr/local/lib

**Comment by layfolk1999** [comment]
> > Hello, @JINGERGERJ. About libglog, there are 3 files in "usr/local/lib", such as libglog.so, libglog.so.0.6.0 and libglog.so.1, the linking order is "libglog.so->libglog.so.1->libglog.so.0.6.0". I removed libglog.so.1 by 'rm' command and linked libglog.so to libglog.so.0.6.0, still there is warning "warning: libglog.so.0, needed by libgici.so, may conflict with libglog.so.1". I entered the command "locate I entered the command "locate libglog | grep usr" and found that libglog.so.1 still exists in the directory "usr/local/lib".
> > <img alt="微信截图_20230712114043" width="1279" src="https://user-images.githubusercontent.com/139234731/252851141-740047ce-3869-4673-b2c1-e4f3c75bdd56.png">
> > Have you encountered this problem?
> 
> Actullay I encountered a similar problem when I try to run the example data, and I shared my solution in my blog:[glog problem](https://blog.csdn.net/weixin_45432823/article/details/131542545) Please check whether you have ever installed the "glog" library using the "apt-get install" command. And the easiest method is to delete all "glog" and "gflags" and re-install them. But you have to watch out for other libraries linking to them. Hope this helps.


Ok, Thank you for your patience, the problem was successfully solved！

**Comment by layfolk1999** [comment]
Ok, Thank you for your patience, the problem was successfully solved ! @zzzzyp-sgg @JINGERGER 
When I installed "glog" and "gflags" before, I did not distinguish the order of installation. And installing "libgoogle-glog-dev" and "libgflags*" related toolkits via "apt-get install" may cause the above problem.

In addition, the previous "segment error" was caused by my installation of a dependency, not an internal problem with the "gici-open" function. Thank you for your answer ! @chichengcn

**Comment by layfolk1999** [comment]
Ok, Thank you for your patience, the problem was successfully solved ! @zzzzyp-sgg @JINGERGER
When I installed "glog" and "gflags" before, I did not distinguish the order of installation. And installing "libgoogle-glog-dev" and "libgflags*" related toolkits via "apt-get install" may cause the above problem.

In addition, the previous "segment error" was caused by my installation of a dependency, not an internal problem with the "gici-open" function. Thank you for your answer ! @chichengcn

**Comment by layfolk1999** [comment]
Ok, Thank you for your patience, the problem was successfully solved ! @zzzzyp-sgg @JINGERGER
When I installed "glog" and "gflags" before, I did not distinguish the order of installation. And installing "libgoogle-glog-dev" and "libgflags*" related toolkits via "apt-get install" may cause the above problem.

In addition, the previous "segment error" was caused by my installation of a dependency, not an internal problem with the "gici-open" function. Thank you for your answer ! @chichengcn
